src/app/clusters/network-commissioning/network-commissioning.h

```diff
@@ -81,6 +83,13 @@ class Instance : public CommandHandlerInterface,
     void SendNonConcurrentConnectNetworkResponse();
 #endif
 
+    // Array for recording NetworkCommissioningInstance.
+    static Instance * NetworkRecord[CHIP_MAX_NETWORK_TYPES];
```

Comment:
Please follow naming conventions for statics; should be something like `sNetworkRecord`.

That said:

1. Why the hard cap at 3?
2. What ensures that these pointers don't dangle if network commissioning instances are dynamically added/destroyed?

It might be better to have Instances live in a linked list of some sort, and remove themselves from it on destruction, etc.

Reply:
Use linked list instead.